docs/proposals/network-policy.md

```diff
@@ -0,0 +1,316 @@
+  // This field is NOT optional and follows standard unversioned.LabelSelector semantics.  
+  // An empty PodSelector matches all pods in this namespace.
+  PodSelector unversioned.LabelSelector `json:"podSelector"`
+
+  // List of ingress rules to be applied to the selected pods.
+  // Traffic is allowed to a pod if Namespace.NetworkPolicy.Ingress.Isolation is undefined, 
+  // OR if the traffic source is the pod's local kubelet (for health checks), 
+  // OR if the traffic matches at least one NetworkPolicyIngressRule across all of the NetworkPolicy 
+  // objects whose podSelector matches the pod.  
+  // If this field is nil, this NetworkPolicy does not affect ingress to the selected pods.
+  // If this field is non-nil but contains no rules, this NetworkPolicy allows no traffic.
+  // If this field is non-nil and contains at least one rule, this NetworkPolicy allows any traffic
+  // which matches at least one of the NetworkPolicyIngressRules in this list.
+  Ingress []NetworkPolicyIngressRule `json:"ingress,omitempty"`
+}
+
+// This NetworkPolicyIngressRule matches traffic if and only if the traffic matches both Ports AND From. 
+type NetworkPolicyIngressRule struct {
+  // List of ports which should be made accessible on the pods selected by PodSelector.  
+  // Each item in this list is combined using a logical OR.  
+  // If this field is nil, this NetworkPolicyIngressRule matches all ports (traffic not restricted by Port). 
+  // If this field is non-nil but contains no items, this NetworkPolicyIngressRule matches no ports (no traffic matches).
+  // If this field is non-nil and contains at least one item, then this NetworkPolicyIngressRule allows traffic 
+  // only if the traffic matches at least one NetworkPolicyPort in the Ports list. 
+  Ports []NetworkPolicyPort `json:"ports,omitempty"`
+
+  // List of sources which should be able to access the pods selected by PodSelector.
+  // Items in this list are combined using a logical OR operation.
+  // If this field nil, this NetworkPolicyIngressRule matches all sources (traffic not restricted by source).
+  // If this field is non-nil but contains no items, this NetworkPolicyIngressRule matches no sources (no traffic matches). 
+  // If this field is non-nil and contains at least on item, this NetworkPolicyIngressRule allows traffic only if the 
+  // traffic matches at least one NetworkPolicyPeer in the From list. 
+  From []NetworkPolicyPeer `json:"from,omitempty"`
+}
+
+type NetworkPolicyPort struct {
+  // The protocol (TCP or UDP) which traffic must match.
+  // If not defined, this field defaults to TCP. 
+  Protocol api.Protocol `json:"protocol"`
+
+  // If specified, the port on the given protocol.  This can 
+  // either be a numerical or named port.  If this field is nil,
+  // this NetworkPolicyPort matches all port names and numbers.
+  // If non-nil, only traffic on the specified protocol AND port
+  // will be matched by this NetworkPolicyPort.
```

Comment:
Are we worried about this being dangerous or error-prone, or are we planning to rely on some schema-validation mechanism to catch field name/indentation typos? By default, a typo like `prot` would cause this field to be silently ignored by the API.